cpp/src/parquet/column_page.h

```diff
@@ -64,33 +65,39 @@ class DataPage : public Page {
   Encoding::type encoding() const { return encoding_; }
   int64_t uncompressed_size() const { return uncompressed_size_; }
   const EncodedStatistics& statistics() const { return statistics_; }
+  std::optional<int64_t> first_row_index() const { return first_row_index_; }
```

Comment:
could we add some docs on when this is expected to be present, what it is (and it is relative to RowGroup or complete file?